storage/flux/table.gen.go

```diff
@@ -288,6 +347,65 @@ func (t *integerTable) advance() bool {
 	return true
 }
 
+// window table
+type integerWindowTable struct {
+	integerTable
+	windowEvery int64
+}
+
+func newIntegerWindowTable(
+	done chan struct{},
+	cur cursors.IntegerArrayCursor,
+	bounds execute.Bounds,
+	every int64,
+	key flux.GroupKey,
+	cols []flux.ColMeta,
+	tags models.Tags,
+	defs [][]byte,
+	cache *tagsCache,
+	alloc *memory.Allocator,
+) *integerWindowTable {
+	t := &integerWindowTable{
+		integerTable: integerTable{
+			table: newTable(done, bounds, key, cols, defs, cache, alloc),
+			cur:   cur,
+		},
+		windowEvery: every,
+	}
+	t.readTags(tags)
+	t.advance()
+
+	return t
+}
+
+func (t *integerWindowTable) advance() bool {
+	a := t.cur.Next()
+	l := a.Len()
+	if l == 0 {
+		return false
+	}
+
+	// Retrieve the buffer for the data to avoid allocating
+	// additional slices. If the buffer is still being used
+	// because the references were retained, then we will
+	// allocate a new buffer.
+	cr := t.allocateBuffer(l)
+	// regain the window start time from the window end time
+	rangeStart := int64(t.bounds.Start)
+	startTimes := make([]int64, len(a.Timestamps))
+	for i := 0; i < a.Len(); i++ {
+		startTimes[i] = a.Timestamps[i] - t.windowEvery
+	}
+	if startTimes[0] < rangeStart {
+		startTimes[0] = rangeStart
+	}
+	cr.cols[startColIdx] = arrow.NewInt(startTimes, t.alloc)
```

Comment:
I'm a bit confused by the code here. It seems to me like the rewrite rule is focused on only pushing down the pattern `window() |> aggregate()`, but the window table implementation seems to be more focused on `aggregateWindow()`. I think this is good because this loop is likely the best way to implement this. We probably don't want to group the table into arrays of size one for performance reasons.

I was expecting to see a piece of code that wrapped this section and use arrow slices to divide this table into multiple tables so, when `window() |> aggregate()` was seen, we would get back tables with the appropriate group keys. Does that idea make sense or am I mistaken?

Reply:
> I was expecting to see a piece of code that wrapped this section and use arrow slices to divide this table into multiple tables so, when window() |> aggregate() was seen, we would get back tables with the appropriate group keys. Does that idea make sense or am I mistaken?

To some extent, yes. We were going to hoist that splitting of single table into multiple tables up to the flux level, and in my mind, that was a separate patch, because this one is already pretty busy. If you don't agree, we can come back and evaluate that.

Reply:
That makes sense to me. I'm concerned about the `PushDownWindowAggregateRule` being included in this patch. Can we comment that back out? It makes sense to me for that to be part of a different PR, but I'm concerned that we'll accidentally set a feature flag to true when the code isn't feature complete.

Reply:
Eh... Actually I am thinking we should do that in the storage... To me it was a mistake I made in the WindowAggregateResultSet

Reply:
My understanding is that if we do it in storage, we send more bytes over the wire.

Reply:
We should discuss that. I much prefer this implementation for storage because it enables us to assemble the result more efficiently on the client side for the various different arrays of `aggregateWindow`. It also dovetails better into a potential future design of aggregate that will be the same as the full `aggregateWindow`. The current design of window which allocates a lot of small tables performs pretty poorly so we want to optimize for the case where someone uses `aggregateWindow` rather than `window() |> aggregate()` specifically. This is also true because `readWindowAggregate() |> duplicate() |> window(every: inf)` will never be as fast as this version and that's going to be the most common query.

Reply:
It's also true we'll send more bytes over the wire because we will send the series header after every row rather than sending it once for an entire series.